dashboard/src/components/AppList/AppList.v2.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from "react";
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import { CdsButton } from "../Clarity/clarity";
+
+import { IFeatureFlags } from "../../shared/Config";
+import { IAppState, IClusterServiceVersion, IResource } from "../../shared/types";
+import * as url from "../../shared/url";
+import Column from "../js/Column";
+import PageHeader from "../PageHeader/PageHeader.v2";
+import SearchFilter from "../SearchFilter/SearchFilter.v2";
+
+import Alert from "components/js/Alert";
+import Row from "components/js/Row";
+import LoadingWrapper from "components/LoadingWrapper/LoadingWrapper.v2";
+import "./AppList.v2.css";
+import AppListGrid from "./AppListGrid";
+
+export interface IAppListProps {
+  apps: IAppState;
+  fetchAppsWithUpdateInfo: (ns: string, all: boolean) => void;
+  cluster: string;
+  namespace: string;
+  pushSearchFilter: (filter: string) => any;
+  filter: string;
+  getCustomResources: (ns: string) => void;
+  customResources: IResource[];
+  isFetchingResources: boolean;
+  csvs: IClusterServiceVersion[];
+  featureFlags: IFeatureFlags;
+  appVersion: string;
+}
+
+function AppList(props: IAppListProps) {
+  const [filter, setFilter] = useState("");
+  const {
+    fetchAppsWithUpdateInfo,
+    filter: filterProps,
+    namespace,
+    getCustomResources,
+    apps: { error, isFetching, listOverview },
+    cluster,
+    isFetchingResources,
+    pushSearchFilter,
+    customResources,
+    appVersion,
+    csvs,
+  } = props;
+
+  useEffect(() => {
+    fetchAppsWithUpdateInfo(namespace, true);
+    getCustomResources(namespace);
+    setFilter(filterProps);
```

Comment:
Are these props something that may change outside of this component? If this is not going to change, you can initialize the state directly. I see `filter` is the one used on the code.

```
  const [filter, setFilter] = useState(props.filter);
```

In addition to this, React recommends you to use an `useEffect` block per behavior. For example, I see here two different blocks:

1. Fetching resources
2. Setting filters

You can split it into two blocks:

```
useEffect(() => {
  fetchAppsWithUpdateInfo(namespace, true);
  getCustomResources(namespace);
}, [namespace, fetchAppsWithUpdateInfo, getCustomResources]);


useEffect(() => {
  setFilter(filterProps);
}, [filterProps]);
```

The purpose of this division is to avoid complex "side effects" that may affect the performance

Reply:
> Are these props something that may change outside of this component?

Yep, it can be changed changing URL.

> You can split it into two blocks:

Nice, didn't know about that :+1: